test/e2e/framework/service/jig.go

```diff
@@ -161,23 +163,10 @@ func (j *TestJig) CreateExternalNameServiceOrFail(namespace string, tweak func(s
 	if err != nil {
 		framework.Failf("Failed to create ExternalName Service %q: %v", svc.Name, err)
 	}
+	j.sanityCheckService(result, svc.Spec.Type)
 	return result
 }
 
-// CreateServiceWithServicePort creates a new Service with ServicePort.
-func (j *TestJig) CreateServiceWithServicePort(labels map[string]string, namespace string, ports []v1.ServicePort) (*v1.Service, error) {
-	service := &v1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: j.Name,
-		},
-		Spec: v1.ServiceSpec{
-			Selector: labels,
-			Ports:    ports,
-		},
-	}
-	return j.Client.CoreV1().Services(namespace).Create(service)
-}
```

Comment:
This we have to keep, otherwise it takes out an essential step from the e2e tests where it is used.
And looking at the e2e tests without this brings out a good point.

If a method creates something, it is great if that method checks that the created object is correct and what we would expect but the sanityChecks fail in the presence of an error.
There is also the fact that `test/e2e/framework/service` is supposed to be a library for users to consume, not "test code" so failing is a bit weird.

I think we should actually rewrite `sanityCheckService()` to return an error, instead of calling any `framework.Fail()`.
The rewrite all the jig create* methods to call `anityCheckService()` within them and return that error for the end-user to handle.

So for example, rewrite the create methods to something like this:
```go
func (j *TestJig) CreateServiceWithServicePort(labels map[string]string, namespace string, ports []v1.ServicePort) (*v1.Service, error) {
	service := &v1.Service{
		ObjectMeta: metav1.ObjectMeta{
			Name: j.Name,
		},
		Spec: v1.ServiceSpec{
			Selector: labels,
			Ports:    ports,
		},
	}
	result, err j.Client.CoreV1().Services(namespace).Create(service)
        if err != nil {
		return result, fmt.Errorsf("Failed to create TCP Service %q: %v", svc.Name, err)
	}
	err = j.sanityCheckService(result, svc.Spec.Type)
        return result, err
}
```